contrib/node/src/python/pants/contrib/node/targets/node_module.py

```diff
@@ -6,18 +6,32 @@
                         unicode_literals, with_statement)
 
 from pants.base.payload import Payload
+from pants.base.payload_field import PrimitiveField
 
 from pants.contrib.node.targets.node_package import NodePackage
 
 
 class NodeModule(NodePackage):
   """A Node module."""
 
-  def __init__(self, sources=None, address=None, payload=None, **kwargs):
+  def __init__(
+    self, sources=None, build_script=None, output_dir='dist',
+    dev_dependency=False, address=None, payload=None, **kwargs):
     """
     :param sources: Javascript and other source code files that make up this module; paths are
                     relative to the BUILD file's directory.
     :type sources: `globs`, `rglobs` or a list of strings
+
```

Comment:
It's a bit confusion because the usage scenario is a bit confusion.  When dev_depedency is set to True, all we really do is not to include this module in class path (or bundlable js product) so that it does not show up in the final bundle.  Its dependent node_module targets are treated separately based on their target definition.  

The reason is sometimes developers writes a build time tool in JS.  They might include this build tool as a dependency.  An example is `contrib/node/examples/src/node/web-component-button`.  It includes `contrib/node/examples/src/node/web-build-tool` as a dependency.  web-build-tool simply invokes webpack with some customized configuration included. The example is simplified.   A more realistic example is a node_module includes a JS tool for asset uploading or template hydration.
I don't think we support this kind of usage in other languages as this is usage pattern is only common for Node developers.